tests/mock_network/mod.rs

```diff
@@ -582,42 +582,53 @@ fn check_section_info_ack() {
     assert_eq!(node_with_sibling_knowledge, expected_all);
 }
 
+// Only carry out this test when using mock_parsec as this will be a heavy duty test to real parsec.
+#[cfg(feature = "mock_parsec")]
 #[test]
-#[ignore] // Will be fixed or removed when proper pruning lands
-fn vote_prune() {
-    // Create a small network to keep all nodes in a single section. Then repeatedly add and remove
-    // nodes (while making sure no split happens) to cause lot of parsec traffic which should make
-    // the parsec graphs to grow sufficiently so prune is triggered and accumulated.
-
-    let max_section_size = 2 * LOWERED_ELDER_SIZE;
-    let steps = 10;
-
+fn carry_out_parsec_pruning() {
+    let init_network_size = 7;
     let network = Network::new(
         NetworkParams {
             elder_size: LOWERED_ELDER_SIZE,
             safe_section_size: LOWERED_ELDER_SIZE,
         },
         None,
     );
-    let mut nodes = create_connected_nodes(&network, max_section_size);
-
-    for _ in 0..steps {
-        if nodes.len() < max_section_size {
-            // Add node
-            let config = NetworkConfig::node().with_hard_coded_contact(nodes[0].endpoint());
-            let node = TestNode::builder(&network).network_config(config).create();
-            nodes.push(node);
-        } else {
-            // Remove node
-            let _ = nodes.pop();
-        }
+    let mut nodes = create_connected_nodes(&network, init_network_size);
+    poll_and_resend(&mut nodes);
 
+    let parsec_versions = |nodes: &Nodes| {
+        nodes
+            .iter()
+            .map(|node| unwrap!(node.inner.elder_state()).parsec_last_version())
+            .collect_vec()
+    };
+
+    let initial_parsec_versions = parsec_versions(&nodes);
+
+    // Keeps polling (which causes parsec gossiping) till trigger a pruning.
+    let max_gossips = 6_000;
+    for i in 0..max_gossips {
         poll_and_resend(&mut nodes);
+
+        let new_parsec_versions = parsec_versions(&nodes);
+        if initial_parsec_versions
+            .iter()
+            .zip(new_parsec_versions.iter())
+            .all(|(vi, vn)| vi < vn)
+        {
+            break;
+        }
+
+        if i == max_gossips - 1 {
+            panic!("Not all nodes carried out pruning");
+        }
     }
 
-    assert!(nodes
-        .iter()
-        .all(|node| unwrap!(node.inner.parsec_prune_accumulated()) > 0));
+    nodes = create_connected_nodes_until_split(&network, vec![1, 1]);
```

Comment:
Is that reasonable? can you call create_connected_nodes_until_split when a network already has node, and why do you do that, we seem to ignore all nodes we had before.

I guess you only wanted verify_invariant_for_all_nodes?

Reply:
Ah, my fault. it is `add_connected_nodes_until_split` shall be called, not `create_connected_nodes_until_split`